### ISSUE TITLE
`@stylistic/unit-case` reads the flag of a bang standing in a value as part of the unit

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#234]
Found while measuring the comment oracle written for the plan of 2026-08-20 against `unit-case`, which reads a unit out of a value the same way.

### What minimal example or steps are needed to reproduce the bug?

A declaration carrying more than one `!important` run, of which PostCSS reads only the last as a flag:

```css
a { b: 1px!important 2px!important; }
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "rules": {
    "@stylistic/unit-case": "upper"
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.css" --config myconfig.json --fix
```

### What did you expect to happen?

The units in capitals and the flags left alone: `a { b: 1PX!important 2PX!important; }`. A keyword is not a unit, and `unit-case` says nothing about one.

### What actually happened?

```css
a { b: 1PX!IMPORTANT 2PX!important; }
```

The flag of the first bang is spelled in capitals. A single `!important` is safe, because PostCSS takes it out of the value into `decl.important` and `decl.raws.important`; every run in front of the last stays in the value, and `postcss-value-parser` reads `1px!important` as one word.

### What is the cause?

`getDimension` splits such a word into a number and everything behind it, so the unit it hands back is `px!important` rather than `px`, and the rule upper-cases the lot. Nothing between the parser and the rule asks whether the text behind the number is a unit at all.

The `lower` option mirrors it, spelling `!IMPORTANT` back down to `!important` and rewriting a declaration nobody asked it to touch.
